async-nats/src/lib.rs

```diff
@@ -837,23 +872,69 @@ impl ConnectionHandler {
 /// Client is a `Clonable` handle to NATS connection.
 /// Client should not be created directly. Instead, one of two methods can be used:
 /// [connect] and [ConnectOptions::connect]
-#[derive(Clone)]
 pub struct Client {
     sender: mpsc::Sender<Command>,
+    errors: Option<mpsc::Receiver<ServerError>>,
```

Comment:
Semantics of option are awkward, don't expose it to client, for now at-least and we'll just do callbacks that are captured by the events closure.

```suggestion
```

Reply:
removed.